api/src/main/java/org/hawkular/inventory/base/spi/CanonicalPath.java

```diff
@@ -113,6 +114,32 @@ public CanonicalPath visitUnknown(Object entity, Void parameter) {
         }, null);
     }
 
+    public Entity toEntity() {
```

Comment:
I know, but it's ok for it's purpose, i.e. to create the right entity type with the right id, which goes to the `new Relationship()` ctor. Any other idea how to de-serialize a relationship from json? Finding the right source/target in the db is imho over-kill.